### DISCUSSION TITLE
Run Gradle task or similar before the PR is created

### DISCUSSION THREAD

**Discussion by JavierSegoviaCordoba** [renovatebot/renovate#12004] [Request Help]
Any time Renovate create a PR, I would like to run a Gradle task which will add to the changelog what it is updating, i.e.:

```
# Changelog

## [Unreleased]

### Added
...

### Fixed
...

### Updated
- dependency org.jetbrains.dokka:dokka-gradle-plugin to v1.5.31

## [1.0.0] - 2021-09-27
...
``` 

I can create easily a task that does that, but I need to Renovate launch it before the PR is created because if I check if the PR is created by Renovate with another action and I run the task changing it and committing (even squashing), Renovate will trigger again removing this new commit generating an infinite loop.

Is there a workaround for this?

**Comment by rarkins** [comment]
If you're self hosting, and you use the full image, then you might be able to achieve this with post upgrade tasks

**Reply by viceice** [reply]
https://docs.renovatebot.com/configuration-options/#gitignoredauthors might also work

**Reply by JavierSegoviaCordoba** [reply]
I am using GitHub app, but @viceice approach can work for me, is it possible to use codeowners in that configuration or I have to hardcode the users?

**Reply by viceice** [reply]
You need to hard code that

**Reply by JavierSegoviaCordoba** [reply]
Thank you for all the info :)

Is it possible to get the list of all packages that are going to be updated? Currently, I have to parse the table in the PR body, if I can list there inside a details block I could parse it easily.

**Reply by rarkins** [reply]
Yes, see `upgrades` in https://docs.renovatebot.com/templates/
